arangod/Network/ConnectionPool.cpp

```diff
@@ -89,11 +87,11 @@ void ConnectionPool::shutdownConnections() {
 
 /// remove unused and broken connections
 void ConnectionPool::pruneConnections() {
-  const auto ttl = std::chrono::milliseconds(_config.idleConnectionMilli + 10);
+  const std::chrono::milliseconds ttl(_config.idleConnectionMilli);
+  auto now = std::chrono::steady_clock::now();
```

Comment:
I think this now needs to be gathered under the read-lock.
Otherwise we collect a now at `t0`, then wait for someone holding the write lock.
This someone will add a lastLeased := t1 with t1 > t0. 
And will afterwards return the lock.
This way this lease will trigger a number underflow when comparing against ttl, and will be flagged as removed